### PR TITLE
feat(rns): allows sorting and filtering by fiat

### LINKED ISSUE #663: Sorting in RNS by Price should use USD
When sorting in the RNS Buy Screen by price, we should be using the **USD** value now that we have multicurrency. It is currently ordering by number but depending on the Currency prices will differ. See example below where it puts as first row the highest amount, but should be the last one. We need to use USD values when sorting by Price.

 image

### PR DESCRIPTION
**Pull request by jurajpiar**
- makes price-filter unit 'USD'
- SUPPORTED_FIAT, BaseCurrency , BaseFiat
-

Resolves https://github.com/rsksmart/rif-marketplace-ui/issues/663
Followed by: https://github.com/rsksmart/rif-marketplace-ui/issues/664
for [RMKT#523](https://rsklabs.atlassian.net/browse/RMKT-523)

### CODE REVIEW

File: src/models/Fiat.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// Abstract fiat + crypto type to extract 'btc' from SupportedFiatSymbol, SupportedTokenSymbol
+export type SupportedFiatSymbol =
+  | SYSTEM_SUPPORTED_FIAT.usd
+// (fiatMulC): Enable for fiat multicurrency
+//   | SYSTEM_SUPPORTED_FIAT.eur
+//   | SYSTEM_SUPPORTED_FIAT.ars
+//   | SYSTEM_SUPPORTED_FIAT.cny
+//   | SYSTEM_SUPPORTED_FIAT.krw
+//   | SYSTEM_SUPPORTED_FIAT.jpy
+//   | 'btc' // temporary: get from SupportedTokenSymbol
```
Comment (farinalvaro) [inline src/models/Fiat.ts:22]: may we remove this comments or create an issue for this?

**Review by itofarina — COMMENTED** [review]
All new code LGTM, I would suggest to remove comments and also squash this PR or edit the commit https://github.com/rsksmart/rif-marketplace-ui/pull/685/commits/2d5f0886661e037b8b98432efcd61d26a583f8e8

File: src/models/Token.ts [hunk 2]
```diff
@@ -9,9 +10,8 @@ export type SupportedTokenSymbol =
     | SYSTEM_SUPPORTED_SYMBOL.rif
     | SYSTEM_SUPPORTED_SYMBOL.rbtc
 
-export type BaseToken = {
-  symbol: SupportedTokenSymbol
-  displayName: string
+// TODO: rename BaseToken to BaseCrypto?
```
[the remark this answers is not in the document]
Comment (jurajpiar) [reply src/models/Token.ts:13]: indeed.

File: src/models/Fiat.ts (hunk 1 above)
Comment (jurajpiar) [reply src/models/Fiat.ts:22]: yeap

**Comment by jurajpiar** [comment]
> I would suggest to remove comments

Yes, however I wanted to discuss them with the team first.

**Comment by jurajpiar** [comment]
> squash this PR or edit the commit 2d5f088

Why? If the message body worries you, we should discuss, as this is aligned with the conventions we use: [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/#specification)

**Comment by farinalvaro** [comment]
@jurajpiar I meant the content of the commit 😄

**Comment by jurajpiar** [comment]
ah right... yea true

**Review by julianmrodri — APPROVED** [review]
lgtm

**Pull request merged**